executor/hash_table.go

```diff
@@ -115,7 +125,7 @@ func (c *hashRowContainer) GetMatchedRowsAndPtrs(probeKey uint64, probeRow chunk
 			return
 		}
 		if !ok {
-			c.stat.probeCollision++
+			atomic.AddInt64(&c.stat.probeCollision, 1)
```

Comment:
Guess this can be a new hotspot. Myabe we can let user control the HashTalbe's bucket size to fix.

Reply:
Actually, the collision should be small probability event